nonlinear_solid_mechanics/include/parameter_handling.h

```diff
@@ -0,0 +1,417 @@
+                          "Number of Newton-Raphson iterations allowed");
+
+        prm.declare_entry("Tolerance force",
+                          "1.0e-9",
+                          Patterns::Double(0.0),
+                          "Force residual tolerance");
+
+        prm.declare_entry("Tolerance displacement",
+                          "1.0e-6",
+                          Patterns::Double(0.0),
+                          "Displacement error tolerance");
+      }
+      prm.leave_subsection();
+    }
+
+    void
+    NonlinearSolver::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Nonlinear solver");
+      {
+        max_iterations_NR = prm.get_integer("Max iterations Newton-Raphson");
+        tol_f             = prm.get_double("Tolerance force");
+        tol_u             = prm.get_double("Tolerance displacement");
+      }
+      prm.leave_subsection();
+    }
+
+
+    struct Time
+    {
+      double delta_t;
+      double end_time;
+      int    output_interval;
+
+      static void
+      declare_parameters(ParameterHandler &prm);
+
+      void
+      parse_parameters(ParameterHandler &prm);
+    };
+
+    void
+    Time::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Time");
+      {
+        prm.declare_entry("End time", "1", Patterns::Double(), "End time");
+
+        prm.declare_entry("Time step size",
+                          "0.1",
+                          Patterns::Double(),
+                          "Time step size");
+
+        prm.declare_entry("Output interval",
+                          "1",
+                          Patterns::Integer(),
+                          "Output interval");
+      }
+      prm.leave_subsection();
+    }
+
+    void
+    Time::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Time");
+      {
+        end_time        = prm.get_double("End time");
+        delta_t         = prm.get_double("Time step size");
+        output_interval = prm.get_integer("Output interval");
+      }
+      prm.leave_subsection();
+    }
+
+
+    struct NewmarkParameters
+    {
+      double beta;
+      double gamma;
+
+      static void
+      declare_parameters(ParameterHandler &prm);
+
+      void
+      parse_parameters(ParameterHandler &prm);
+    };
+
+    void
+    NewmarkParameters::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Newmark parameters");
+      {
+        prm.declare_entry("beta", "0.25", Patterns::Double(0, 0.5), "beta");
+
+        prm.declare_entry("gamma", "0.5", Patterns::Double(0, 1), "gamma");
+      }
+      prm.leave_subsection();
+    }
+
+    void
+    NewmarkParameters::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Newmark parameters");
+      {
+        beta  = prm.get_double("beta");
+        gamma = prm.get_double("gamma");
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    struct PreciceConfiguration
```

Comment:
```suggestion
    struct PreciceAdapterConfiguration
```
to not mix it up with `precice-config-xml` ?